sale_commission_product_criteria/models/commission.py

```diff
@@ -0,0 +1,392 @@
+# © 2023 ooops404
+# License AGPL-3 - See https://www.gnu.org/licenses/agpl-3.0.html
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+from odoo.tools import float_repr
+
+COMMISSION_ITEM_TYPES = {
+    "0_product_variant": 1000,
+    "1_product": 10000,
+    "2_product_category": 100000,
+    "3_global": 1000000,
+}
+
+
+class SaleCommissionLineMixin(models.AbstractModel):
+    _inherit = "sale.commission.line.mixin"  # OCA sale_commission
```

Comment:
Don't mix this model in the same file, as they are not directly related by a o2m relation.